micronota/db/tigrfam.py

```diff
@@ -0,0 +1,236 @@
+       This field may contain more than one EC number, with a single space as
+       the separator.
+   EN  Expanded name - A fuller or informative alternate version of the
+       definition line. This compares (or is identical) to the more terse DE
+       definition that is used for automated annotation.
+   TP  Always "TIGRFAMs", the identifier of this database
+   === ========================================================================
+
+Proteins that score
+above the trusted cutoffs are believed to reside within the family and
+those falling below the noise cutoffs are believed to reside outside the
+family.  The margin of error with respect to presence or absence of a
+protein within a TIGRFAMs family is represented by the score range
+between noise and trusted cutoffs.
+
+Because the number of completed and nearly completed genomes
+has now entered the thousands, protein families are becoming
+very large, and exceptions may be found in certain equivalog
+families.  Exceptions usually represent neofunctionalizations
+that arise within an equivalog family, although some may
+represent paralogs that are minimally derived since their
+branching from the equivalog familiy (i.e. short branch length).
+
+TIGRFAMs now includes 57 models of type "exception",  a type
+of model that overrules annotation from an equivalog model,
+either to give more specific information or to correct annotation
+for a neofunctionalized subgroup.
+
+TIGRFAMS_GO_LINK
+^^^^^^^^^^^^^^^^
+Gene Ontology (GO) term assignments.
+Models may have  0, 1, or several GO assignments.
+
+
+Reference
+---------
+.. [1] http://www.ncbi.nlm.nih.gov/pubmed/12520025
+.. [2] http://www.ncbi.nlm.nih.gov/pubmed/23197656
+
+'''
+
+
+# ----------------------------------------------------------------------------
+# Copyright (c) 2015--, micronota development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import os
+import shutil
+import gzip
+import tarfile
+from tempfile import mkdtemp
+from os.path import join
+from ftplib import FTP
+from sqlite3 import connect
+
+from ..bfillings.hmmer import hmmpress_hmm
+
+
+def prepare_db(out_d, prefix='tigrfam_v15.0'):
+    '''Download and prepare TIGRFAM database.
+
+    Parameters
+    ----------
+    out_d : str
+        The directory of output files.
+    prefix : str
+        The file name (without extensions) of the output files.
+    '''
+    server = 'ftp.tigr.org'
+    path = 'pub/data/TIGRFAMs'
+
+    hmm = 'TIGRFAMs_15.0_HMM.LIB.gz'
+    hmm_out = join(out_d, '%s.hmm' % prefix)
+
+    metadata = 'TIGRFAMs_15.0_INFO.tar.gz'
+    metadata_out = join(out_d, '%s.db' % prefix)
+
+    try:
+        temp_dir = mkdtemp()
+        hmm_tmp = join(temp_dir, hmm)
+        metadata_tmp = join(temp_dir, metadata)
+        print(temp_dir)
```

Comment:
oh, should be deleted.

On Thu, Jan 21, 2016 at 7:46 PM, mortonjt <notifications@github.com> wrote:

> In micronota/db/tigrfam.py
> <https://github.com/biocore/micronota/pull/40#discussion_r50499569>:
>
> > +        The file name (without extensions) of the output files.
> > +    '''
> > +    server = 'ftp.tigr.org'
> > +    path = 'pub/data/TIGRFAMs'
> > +
> > +    hmm = 'TIGRFAMs_15.0_HMM.LIB.gz'
> > +    hmm_out = join(out_d, '%s.hmm' % prefix)
> > +
> > +    metadata = 'TIGRFAMs_15.0_INFO.tar.gz'
> > +    metadata_out = join(out_d, '%s.db' % prefix)
> > +
> > +    try:
> > +        temp_dir = mkdtemp()
> > +        hmm_tmp = join(temp_dir, hmm)
> > +        metadata_tmp = join(temp_dir, metadata)
> > +        print(temp_dir)
>
> Do you want this?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/biocore/micronota/pull/40/files#r50499569>.
>